### PR TITLE
Feat(Task-7): As a User, I want a minimalistic StartPage.

### PR DESCRIPTION
**Pull request by SamuFigueirido**

### CODE REVIEW

File: src/app/app.currentWeather.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export default {};
```
Comment (eloydotco) [inline src/app/app.currentWeather.js]: `app.` prefix is not necessary, that prefix format is used only for component related files.
a better name could be `currentWeather.value.js` or other options is moved all values for a folder called values and the name can be just `currentWeather.js`.
We have to pursuit intuitive names and paths in order to find everything asap :)

File: src/app/app.lastSearches.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export default {};
```
Comment (eloydotco) [inline src/app/app.lastSearches.js]: same as before, `app.` prefix is not necessary...

File: src/app/app.lastSearchesService.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+export default class LastSearchesService {
```
Comment (eloydotco) [inline src/app/app.lastSearchesService.js]: Same as before, `app.` prefix is not necessary (`lastSearch.service.js`, for example). In this case, a folder called services can be interesting :)

File: src/app/app.lastSearchesService.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    constructor(lastSearches) {
+        this.lastSearches = lastSearches;
+    }
+
+    saveCity(city) {
+        if (city === undefined || city.trim() === '') {
+            return;
+        } else {
```
Comment (diegopf) [inline src/app/app.lastSearchesService.js]: If you are returning in the if block you don't have to wrap your logic in the else block

File: src/app/app.lastSearchesService.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+            const cityAux = (city.charAt(0).toUpperCase() + city.slice(1).toLowerCase()).trim();
```
Comment (diegopf) [inline src/app/app.lastSearchesService.js]: why is this needed?

File: src/app/app.openWeatherMapsService.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+export default class OpenWeatherMapService {
+    constructor($http) {
+        this.$http = $http;
+    }
+    getCurrentWeather(city) {
+        const apiKey = '608c0e7fd9bce774dd414deb10de6e52';
```
Comment (eloydotco) [inline src/app/app.openWeatherMapsService.js]: In this case is not necessary, but for the future is better to have apiKey constant available for the whole service (just moving this line to the top of the file)

File: src/app/app.openWeatherMapsService.js [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+        const request = {
+            method: 'GET',
+            url: 'http://api.openweathermap.org/data/2.5/weather?q=' + city+ '&appid=' + apiKey,
```
Comment (eloydotco) [inline src/app/app.openWeatherMapsService.js]: Please, use template literals
https://css-tricks.com/template-literals/#article-header-id-1

File: src/app/app.lastSearchesService.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+            if (this.lastSearches.includes(cityAux)) {
+                return;
+            } else {
+                this.lastSearches.unshift(cityAux);
+                if (this.lastSearches.length > 10) {
```
Comment (diegopf) [inline src/app/app.lastSearchesService.js]: wouldn't be easier just to a this.lastSearches.pop() ?

**Review by eloydotco — CHANGES_REQUESTED** [review]

File: src/app/app.openWeatherMapsService.js (hunk 6 above)
Comment (diegopf) [inline src/app/app.openWeatherMapsService.js]: You must not commit your secrets into a public repository.

See https://github.com/jLagosCarrera/MeteoApp/pull/4#discussion_r280689452 

You can ask Julio about how to solve this

**Review by diegopf — CHANGES_REQUESTED** [review]

File: src/app/app.lastSearchesService.js (hunk 5 above)
Comment (SamuFigueirido) [reply src/app/app.lastSearchesService.js]: To save and show the names of the cities with the first letter in capital, and the rest with lower case

File: src/app/app.module.js [hunk 9]
```diff
@@ -3,8 +3,18 @@ import uirouter from '@uirouter/angularjs';
...
+import startPage from './start-page/start-page';
+import lastSearches from './values/lastSearches';
+import currentWeather from './values/currentWeather';
+import "material-icons/iconfont/material-icons.scss";
+import "angular-material/angular-material.scss";
 
 export default angular
-    .module('app', [uirouter, ngMaterial, ngAnimate, ngAria])
-    .factory('weatherService', weatherService);
+    .module('startpage', [uirouter, ngMaterial, ngAnimate, ngAria])
```
Comment (gbtec-ext-jorgevillar) [inline src/app/app.module.js]: the 'uirouter, ngMaterial, ngAnimate, ngAria' will be used in all the project, I would maintain the 'app' module as a parent and import there this dependencies and your start page module

File: src/app/services/lastSearches.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
+export default class LastSearchesService {
+    constructor(lastSearches) {
+        this.lastSearches = lastSearches;
+    }
+
+    saveCity(city) {
+        if (city === undefined || city.trim() === '') {
+            return;
+        } else {
```
Comment (gbtec-ext-jorgevillar) [inline src/app/services/lastSearches.js]: if you do a return inside the if you dont need the else

File: src/app/start-page/start-page.scss [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+  color: $salmon;
+}
+md-input-container:not(.md-input-invalid).md-input-focused .md-input {
+  border-color: $salmon;
+}
+md-input-container:not(.md-input-invalid) .md-input {
+  border-color: rgb(172, 172, 172);
+}
+md-input-container:not(.md-input-invalid).md-input-focused label {
+  color: $salmon;
```
Comment (gbtec-ext-jorgevillar) [inline src/app/start-page/start-page.scss]: You can join this style with the '.salmon' and the 'md-input-container:not(.md-input-invalid) .md-input'

File: src/app/start-page/start-page.scss [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+  list-style-type: none;
+  text-align: center;
+  border: 2px solid $salmon;
+  background-color: $salmon;
+  color: white;
+  box-shadow: 0px 2px 5px rgba(0, 0, 0, 0.26);
+  margin-bottom: 10px;
+  margin-top: 10px;
+  margin-right: 25px;
+  margin-left: 25px;
```
Comment (gbtec-ext-jorgevillar) [inline src/app/start-page/start-page.scss]: Instead of do this in four lines you can do it like this:
`margin: 10px 25px;`  (margin: <verticalMargin>, <horizontalMargin>)
You jhave this override too: 
`margin: <top>, <right>, <bottom>, <left>`

File: src/app/start-page/start-page.html [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+        <h1 class="skyblue">a</h1>
+        <h1 class="brown">t</h1>
+        <h1 class="salmon">h</h1>
+        <h1 class="skyblue">e</h1>
+        <h1 class="brown">r</h1>
+        <a class="reg salmon">™</a>
+    </div>
+    <form autocomplete="off" layout="row" layout-align="center center" ng-submit="$ctrl.searchCity()">
+        <md-input-container class="container">
+            <label>{{isSpanish ? "Mostrarme el tiempo en..." : "Show me the weather in..."}}</label>
```
Comment (gbtec-ext-jorgevillar) [inline src/app/start-page/start-page.html]: Use single quotes for JS stuff

**Review by gbtec-ext-jorgevillar — CHANGES_REQUESTED** [review]

File: src/app/services/lastSearches.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+    saveCity(city) {
+        if (city === undefined || city.trim() === '') {
+            return;
+        }
+        const cityAux = city.trim().toLowerCase();
+        if (this.lastSearches.includes(cityAux)) {
+            return;
+        }
+        this.lastSearches.unshift(cityAux);
+        if (this.lastSearches.length === 11) {
```
Comment (diegopf) [inline src/app/services/lastSearches.js]: wouldn't be better check if length is bigger than 10?

**Review by diegopf — CHANGES_REQUESTED** [review]

**Review by diegopf — APPROVED** [review]

File: src/app/services/lastSearches.js [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+        if (city === undefined || city.trim() === '') {
+            return;
+        }
+        const cityAux = city.trim().toLowerCase();
+        if (this.lastSearches.includes(cityAux)) {
+            return;
+        }
+        this.lastSearches.unshift(cityAux);
+        if (this.lastSearches.length > 10) {
+            this.lastSearches.pop();
```
Comment (eloydotco) [inline src/app/services/lastSearches.js:16]: Since the condition is a length bigger than 10, maybe is a good idea to get only the first 10 elements, instead of remove the last

**Review by eloydotco — APPROVED** [review]

**Review by gbtec-ext-jorgevillar — APPROVED** [review]
